src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
+    private static final int INDEX_OF_DATE = 0;
+    private static final int INDEX_OF_NAME = 1;
+    private static final int INDEX_OF_HOURS = 2;
+    private static final int INDEX_OF_RATE = 3;
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period ");
+        builder.append(dateFrom).append(" - ").append(dateTo);
+
+        String[] filteredData = filterDataByDate(data, dateFrom, dateTo);
+        for (String name : names) {
+            builder.append(System.lineSeparator()).append(name).append(" - ");
+            int countedMoney = 0;
+            for (String input : filteredData) {
+                if (input != null && name.equals(input.split(" ")[INDEX_OF_NAME])) {
+                    countedMoney += Integer.parseInt(input.split(" ")[INDEX_OF_HOURS])
+                                   * Integer.parseInt(input.split(" ")[INDEX_OF_RATE]);
+                }
+            }
+            builder.append(countedMoney);
+        }
+        return builder.toString();
+    }
+
+    private String[] filterDataByDate(String[] data, String dateFrom, String dateTo) {
```

Comment:
better delete this method and simply add condition to IF on line 24
Use methods `isBefore`, `isAfter` - no need to use Period